examples/Transfer_Learning/basic_transfer_learning.py

```diff
@@ -0,0 +1,175 @@
+# per input dimension of the test function:
+
+discrete_params = [
+    NumericalDiscreteParameter(
+        name=f"x{d}",
+        values=np.linspace(lower, upper, POINTS_PER_DIM),
+    )
+    for d, (lower, upper) in enumerate(BOUNDS.T)
+]
+
+# ```{note}
+# While we could optimize the function using
+# [NumericalContinuousParameters](baybe.parameters.numerical.NumericalContinuousParameter),
+# we use discrete parameters here because it lets us interpret the percentages shown in
+# the final plot directly as the proportion of candidates for which there were target
+# values revealed by the training function.
+# ```
+
+# Next, we define a
+# [TaskParameter](baybe.parameters.categorical.TaskParameter) to encode the task context,
+# which allows the model to establish a relationship between the training data and
+# the data collected during the optimization process.
+# Because we want to obtain recommendations only for the test function, we explicitly
+# pass the `active_values` keyword.
+
+task_param = TaskParameter(
+    name="Function",
+    values=["Test_Function", "Training_Function"],
+    active_values=["Test_Function"],
+)
+
+# With the parameters at hand, we can now create our search space.
+
+parameters = [*discrete_params, task_param]
+searchspace = SearchSpace.from_product(parameters=parameters)
+
+# (Lookup)=
+### Generating Lookup Tables
+
+# We generate two lookup tables containing the target values of both test
+# functions at the given parameter grid.
+# Parts of one lookup serve as the training data for the model.
+# The other lookup is used as the loop-closing element, providing the target values of
+# the test functions on demand.
+
+grid = np.meshgrid(*[p.values for p in discrete_params])
+
+lookups: Dict[str, pd.DataFrame] = {}
+for function_name, function in test_functions.items():
+    lookup = pd.DataFrame({f"x{d}": grid_d.ravel() for d, grid_d in enumerate(grid)})
+    lookup["Target"] = lookup.apply(function, axis=1)
+    lookup["Function"] = function_name
+    lookups[function_name] = lookup
+lookup_training_task = lookups["Training_Function"]
+lookup_test_task = lookups["Test_Function"]
+
+### Simulation Loop
+
+# We now simulate campaigns for different amounts of training data unveiled,
+# to show the impact of transfer learning on the optimization performance:
+
+results: List[pd.DataFrame] = []
+for p in (0.01, 0.02, 0.05, 0.08, 0.2):
+    campaign = Campaign(searchspace=searchspace, objective=objective)
+    initial_data = [lookup_training_task.sample(frac=p) for _ in range(N_MC_ITERATIONS)]
+    result_fraction = simulate_scenarios(
+        {f"{100*p}": campaign},
+        lookup_test_task,
+        initial_data=initial_data,
+        batch_size=BATCH_SIZE,
+        n_doe_iterations=N_DOE_ITERATIONS,
+    )
+    results.append(result_fraction)
+
+# For comparison, we also optimize the function without using any initial data:
+
+result_fraction = simulate_scenarios(
```

Comment:
a strange name, suggest `result_baseline` or `result_no_tl`

Reply:
I will probably remove this part anyway if this is fine for you. I think it is really ugly that we need this additional call, and I also think that the demonstration of this being better than the baseline is not necessary since we just want to demonstrate how to use it here. I'd rather include some plot that shows improvements on a smaller example in the user guide (without the code, just as a demonstration) if that is gucci (but we can also not do this at all)

Reply:
fine but that would rbeak with the idea that these examples now create the polished pictures we use in user guides?

Reply:
I thought this is the working mode: this (and other upcoming) examples now produce detailed plots, which are reused in userguides. The userguides do not use plots that are created somewhere else that is not published (clearly this doesnt apply to schematics).

But having the baseline is important for the userguide. So if you remove it this picture cannot be used anymore in the userguide.

Which opens the question where does the userguide picture come from.

Reply:
You are right, I simply forgot that this was what we agreed to do. So I'll keep this part and rename accordingly.

Reply:
Changed to `results_baseline`.